src/client.ts

```diff
@@ -301,7 +301,7 @@ class Client {
         // Handle native or bluebird Promise rejections
         // https://developer.mozilla.org/en-US/docs/Web/Events/unhandledrejection
         // http://bluebirdjs.com/docs/api/error-management-configuration.html
-        let reason = (<PromiseRejectionEvent>e).reason || (<CustomEvent>e).detail.reason;
+        let reason = (<PromiseRejectionEvent>e).reason || (<CustomEvent>e).detail && (<CustomEvent>e).detail.reason || 'unhandled rejection with no reason given';
```

Comment:
This is a bit too smart now but I guess we can decipher it later.

Reply:
@bruderstein I have a questions since you said you were facing too many `Error: Uncaught TypeError: Cannot read property 'reason' of undefined` errors. But, since this fix aren't these replaced with `unhandled rejection with no reason given` errors ?

Reply:
@vmihailenco I would suggest to set the `reason` as `unhandled rejection with no reason given` if the reason is empty so that the stack trace is preserved. This solution just resets the stacktrace and all I can see is this,
<img width="981" alt="screenshot 2019-02-01 at 5 00 47 pm" src="https://user-images.githubusercontent.com/1897178/52121825-ec90f580-2646-11e9-8ae7-d55158cee18d.png">
Not very helpful.

Reply:
I don't follow how that should preserve backtrace - could you explain your suggestion?

Reply:
@vmihailenco Sure. So, if `PromiseRejectionEvent` object doesn't have a reason, we are returning a string and that results in the ☝️ stack.

But, instead as a developer, I would like to know where exactly the promise was rejected and if I could handle those rejections gracefully. I was wondering if we can see that stack from the `PromiseRejectionEvent` object.

Reply:
AFAIK stack is not available in `PromiseRejectionEvent`.

Reply:
@vmihailenco In that case, do we have an option to stop tracking "Unhandled Rejection"?

Reply:
Yes, you can add a filter - https://github.com/airbrake/airbrake-js#filtering-errors

Reply:
@vmihailenco Thanks. That was really helpful. 👍